coverage_point_calculator/src/lib.rs

```diff
@@ -0,0 +1,933 @@
+    pub covered_hexes: Vec<CoveredHex>,
+    pub boosted_hex_eligibility: BoostedHexStatus,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("signal level {0:?} not allowed for {1:?}")]
+    InvalidSignalLevel(SignalLevel, RadioType),
+}
+
+pub fn calculate_coverage_points(radio: RewardableRadio) -> CoveragePoints {
+    let hex_coverage_points = radio.covered_hexes.calculated_coverage_points();
+    let location_trust_multiplier = radio.location_trust_scores.multiplier;
+    let speedtest_multiplier = radio.speedtests.multiplier;
+
+    let coverage_points = hex_coverage_points * location_trust_multiplier * speedtest_multiplier;
+    let total_coverage_points = coverage_points.round_dp_with_strategy(2, RoundingStrategy::ToZero);
+
+    CoveragePoints {
+        total_coverage_points,
+        hex_coverage_points,
+        location_trust_multiplier,
+        speedtest_multiplier,
+        // Radio information
+        radio_type: radio.radio_type,
+        radio_threshold: radio.radio_threshold,
+        speedtests: radio.speedtests.speedtests,
+        location_trust_scores: radio.location_trust_scores.trust_scores,
+        covered_hexes: radio.covered_hexes.hexes,
+        boosted_hex_eligibility: radio.boosted_hex_eligibility,
+    }
+}
+
+impl RewardableRadio {
+    pub fn new(
+        radio_type: RadioType,
+        speedtests: Vec<Speedtest>,
+        location_trust_scores: Vec<LocationTrust>,
+        radio_threshold: RadioThreshold,
+        covered_hexes: Vec<RankedCoverage>,
+    ) -> Result<Self> {
+        // QUESTION: we need to know about boosted hexes to determine location multiplier.
+        // The location multiplier is then used to determine if they are eligible for boosted hexes.
+        // In the case where they cannot use boosted hexes, should the location mulitiplier be restored?
+
+        let any_boosted_hexes = covered_hexes.iter().any(|hex| hex.boosted.is_some());
+        let location_trust_scores = if any_boosted_hexes {
+            LocationTrustScores::new_with_boosted_hexes(&radio_type, location_trust_scores)
+        } else {
+            LocationTrustScores::new(&radio_type, location_trust_scores)
+        };
+
+        let boosted_hex_status = BoostedHexStatus::new(
+            &radio_type,
+            location_trust_scores.multiplier,
+            &radio_threshold,
+        );
+
+        let covered_hexes = if boosted_hex_status.is_eligible() {
+            CoveredHexes::new(&radio_type, covered_hexes)?
+        } else {
+            CoveredHexes::new_without_boosts(&radio_type, covered_hexes)?
+        };
+
+        Ok(Self {
+            radio_type,
+            speedtests: Speedtests::new(speedtests),
+            location_trust_scores,
+            radio_threshold,
+            covered_hexes,
+            boosted_hex_eligibility: boosted_hex_status,
+        })
+    }
+}
+
+impl CoveragePoints {
+    pub fn iter_boosted_hexes(&self) -> impl Iterator<Item = CoveredHex> {
```

Comment:
who/what is the expected caller of this ? currently unused

Reply:
I believe the idea is that this will be used in the mobile-verifier to help generate the `boosted_hexes` field the `radio_reward`

https://github.com/helium/proto/blob/fdb83a38fbe8aead4ff6cb39d1e996f0ad0646b7/src/service/poc_mobile.proto#L438

Reply:
Brian is correct. I'm going to remove this function for now, as I think the updates that are still ongoing will change how boosted hexes will be gotten by a caller, and we can put that functionality back in when it's needed.